common/retry.go

```diff
@@ -53,10 +53,18 @@ func IsRetryableKubeAPIError(err error) bool {
 func Convert2WaitBackoff(backoff *apicommon.Backoff) (*wait.Backoff, error) {
 	result := wait.Backoff{}
 
-	if backoff.Duration != nil {
-		result.Duration = time.Duration(backoff.Duration.Int64Value())
+	d := backoff.Duration
+	if d == nil {
+		d = &defaultDuration
+	}
+	if d.Type == apicommon.Int64 {
+		result.Duration = time.Duration(d.Int64Value())
 	} else {
-		result.Duration = time.Duration(defaultDuration.Int64Value())
+		parsedDuration, err := time.ParseDuration(d.StrVal)
```

Comment:
suggestion - https://github.com/argoproj/pkg/blob/master/env/env.go

Reply:
It does not lookup env here, but that's a good toolkit.